### ISSUE TITLE
[RFC] Sharing status indicator

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/server#43443]
Hi @nextcloud/designers, we've had a few changes on the sharing status indicator and I've seen various changes requests or small sentences here and there that lead me to ask what you have in mind really.

### From https://github.com/nextcloud/server/issues/41620
Current | Medium with avatars + icons | Remove label for shares from others
-|-|-
![Sharing current](https://github.com/nextcloud/server/assets/925062/e475d9ae-3820-468e-92b1-7f1b0b7eda5c)|![sharing medium](https://github.com/nextcloud/server/assets/925062/f72a03e0-f686-425e-be56-c2358a65ea33)|![Sharing plain](https://github.com/nextcloud/server/assets/925062/a6276772-405e-4b8f-9ea9-465664bcfce0)

> @jancborchardt said
> We discussed it with @nextcloud/designers and will go with the middle option as that is a good middle ground between the 2, and it takes into account Clémentine’s feedback from user testing.

### From https://github.com/nextcloud/server/pull/41656
> @marcoambrosini said
> For what is worth, I also think it's better to always show the shared label at all times. I think it's the 2nd most important thing after the file name. Let's figure out a better strategy to make things clear soon!

### Other references:
- Orignal draft: https://github.com/nextcloud/server/issues/38156

### Current implementation
If I share a file, even multiple times with different shares, we use the following logic (was like that in old times too)

| Share type | Icon |
|:---|:--:|
| `link` or `email` |🔗|
| `group` or `federated group` |👩‍👧‍👦|
|`circles` |⭕|
|not shared| 🧑⁺|

## Final request
As far as I understood, and this is where I need confirmation:

|Owner|Shared with|Result|
|:--|:--|--:|
|mine|none|![image](https://github.com/nextcloud/server/assets/14975046/426de4d1-4de9-494f-9787-8c86c4138724)|
|mine|`link` or `email`|![image](https://github.com/nextcloud/server/assets/14975046/28f32a66-4ab0-4f95-9534-57ce6a0a9e5c)|
|mine|`group` or `federated group`|![image](https://github.com/nextcloud/server/assets/14975046/9b19f58f-9100-45b2-864b-b5a128072451)|
|mine|`user`|![image](https://github.com/nextcloud/server/assets/14975046/8578c8f9-f6fa-414d-97ad-e3b0889897e4)|
|someone else|none|![image](https://github.com/nextcloud/server/assets/14975046/0c92cefe-ed15-4ee3-ba3f-b2be854db816)|
|someone else|`link` or `mail`|![image](https://github.com/nextcloud/server/assets/14975046/9c39a58a-beac-484d-946a-9c1a3e97c8f6)|
|(you get the point for the Shared with icons)|any|owner-avatar + matching type icon that will be primary|

:warning:  If we have multiple shares, we use the logic mentioned above: mail/link, then group/federated group, then circles and finally user

It seems to be what I understood, please confirm so I can fix the code made in a hurry for 28! :pray:

**Comment by jancborchardt** [comment]
@skjnldsv so the current state was the result of discussions of @karlitschek and me, this is why it is like it is.

As to your specific questions – the first 4 entries look good.

However the shares from someone else don’t look like you show, but rather like this, which is as expected and designed:
![image](https://github.com/nextcloud/server/assets/925062/fbb9f233-e45e-4ee7-868f-2199bf6c1087)

The only strangeness being the design folder which is shared with both groups and accounts, so the icons of the folder and the share action don’t line up (should both be the "group" icon by the logic you describe, right?):
![image](https://github.com/nextcloud/server/assets/925062/cd5d78dc-042a-4ce9-9413-b60f356e362c)

**Comment by skjnldsv** [comment]
Alright, seems like I'm missing some internal discussion :thinking: 

Your messages from https://github.com/nextcloud/server/issues/41620 seems to say you wanted to go for [the middle solution](https://github.com/nextcloud/server/assets/14975046/2f031356-7eb6-4398-a559-d272888953bf)?
Which seems to show both avater and sharing icon?
This is the entire reason of me creating this thread. According to Marco's comments, there was no time to fully implement it for 28

Shall I assume you changed your mind and are ok keeping the [the last one](https://github.com/nextcloud/server/assets/14975046/381c206b-149a-43ca-b698-dabe9f75311c) ?

I just wanted to avoid missing some stuff that seemed spread over multiple comments that the design team wanted :pray:

**Comment by jancborchardt** [comment]
@skjnldsv right, so from the mockups from https://github.com/nextcloud/server/issues/41620 for both @karlitschek and me both the 2 options on the right would work.

The middle option was never implemented and rather converted into the left option which is too much.

But since we had feedback that the option with only icons can lead to misunderstanding, not finding the share option, and considering both @karlitschek and I would be good with the middle solution – let’s try that out at least as a prototype, and possibly for 29.

During development with full text everywhere | Medium with avatars + icons | Current (Remove label for shares from others)
-|-|-
![Sharing current](https://github.com/nextcloud/server/assets/925062/e475d9ae-3820-468e-92b1-7f1b0b7eda5c)|![sharing medium](https://github.com/nextcloud/server/assets/925062/f72a03e0-f686-425e-be56-c2358a65ea33)|![Sharing plain](https://github.com/nextcloud/server/assets/925062/a6276772-405e-4b8f-9ea9-465664bcfce0)

**Comment by skjnldsv** [comment]
Looking good!
Let's try to test this with NC30 and see if it still works out :)

**Comment by skjnldsv** [comment]
@jancborchardt is the middle solution still wanted?
![sharing medium](https://github.com/nextcloud/server/assets/925062/f72a03e0-f686-425e-be56-c2358a65ea33)

**Comment by jancborchardt** [comment]
@skjnldsv yes, as per my last comment above that is what would still be wanted, and we also had usability tests validate this. (People used the action menu rather than clicking on the avatar when wanting to share an already shared file.

**Comment by ornanovitch** [comment]
is this going to be tackled with #55428 ?
